src/main/java/SnippetManager.java

```diff
@@ -0,0 +1,120 @@
+            }
+        }
+    }
+
+    private String generatePathToSnippetJson(Integer snippetId) {
+        return pathToSnippoDir + "/" + snippetId;
+    }
+
+    String listSnippets(Map<Integer, Snippet> snippetsToList) {
+        StringBuilder response = new StringBuilder("All snippetsToList: \n");
+
+        for (Integer snippetId: snippetsToList.keySet()) {
+            response.append("Id: "+snippetId+"\n").append(snippetsToList.get(snippetId).toString()).append("\n");
+        }
+
+        return response.toString();
+    }
+
+    String listAll() {
+        return listSnippets(snippets);
+    }
+
+    Integer create(String title, String content, String language, String[] tags) {
+        Integer newSnippetId = getNextId();
+        snippets.put(
+                newSnippetId,
+                new Snippet(generatePathToSnippetJson(newSnippetId), title, content, language, tags)
+        );
+
+        return newSnippetId;
+    }
+
+    String read(Integer id) {
+        if (isValidId(id)) {
+            return snippets.get(id).toString();
+        } else {
+            onSnippetNotFound();
+            return "";
+        }
+    }
+
+    void delete(Integer id) {
+        if (isValidId(id)) {
+            if (new File(generatePathToSnippetJson(id)).delete()) {
+                snippets.remove(id);
+            } else {
+                System.out.println("Cannot delete the snippet and its file. Id: " + id);
+            }
+        } else {
+            onSnippetNotFound();
+        }
+    }
+
+    void edit(Integer id) {
+        if (isValidId(id)) {
+            Snippet snippetToEdit = snippets.get(id);
+            snippetToEdit.edit();
+        } else {
+            onSnippetNotFound();
+        }
+    }
+
+    Map<Integer, Snippet> filter(String wordToContain, String language, String[] tags) {
+        return snippets.entrySet()
+                .stream()
+                .filter(
+                        entry -> entry.getValue().getTitle().contains(wordToContain) ||
+                                entry.getValue().getContent().contains(wordToContain)
+                )
+                .filter(entry -> language.equals("") ||
+                        entry.getValue().getLanguage().equals(language))
+                .filter(entry -> (tags.length == 1 && tags[0].equals("")) ||
+                        Arrays.asList(entry.getValue().getTags()).containsAll(Arrays.asList(tags)))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    // Returns the next available Id for a new snippet
+    private int getNextId() {
+        return snippets.isEmpty() ?
+                1 : Collections.max(snippets.keySet())+1;
+    }
+
+    private boolean isValidId(Integer id) {
+        return snippets.containsKey(id);
+    }
+
+    private void onSnippetNotFound() {
+        System.out.println("Snippet not found");
+    }
+
+    HashMap<Integer, Snippet> getSnippets() {
+        return snippets;
+    }
+}
```

Comment:
Good code. Not difficult to read/understand, used exception handling, clear names for functions and variables, no repeated functions, functions do as their names say and are not too long.